### ISSUE TITLE
Loss not decreasing when training outside of script

### ISSUE THREAD

**Issue by plehman2000** [rstrudel/segmenter#59]
After running 
`python -m segm.scripts.prepare_ade20k $DATASET`
`python -m segm.train --log-dir seg_tiny_mask --dataset ade20k \
  --backbone vit_tiny_patch16_384 --decoder mask_transformer `

The training script runs without error, but the loss is not decreasing:
![image](https://user-images.githubusercontent.com/54288924/227565539-8fc4c3bc-ca37-4a36-84f2-f4adfd8fde64.png)

Upon further examination, when I isolated your segmenter class, I found that the parameters after the decoders transformer step do not seem to be updating. 

See my note in the comment below for clarification. Any help is appreciated! I've been stuck for quite a while on this

(from decoder.py, MaskTransformer())
![image](https://user-images.githubusercontent.com/54288924/227567194-f33793f8-091f-4f02-8762-37d7f9d35844.png)

**Comment by plehman2000** [comment]
For clarification, I see that the average loss is decreasing, but is there any reason that making a minimum working example like this would not yield a changing loss?
# MWE
![image](https://user-images.githubusercontent.com/54288924/227569067-e027e53a-92e7-4721-b110-4b8248cb0dce.png)
![image](https://user-images.githubusercontent.com/54288924/227569123-45e2dff2-efee-4ca0-a178-1bb65dabf1a5.png)

# Results
## Unedited model
![image](https://user-images.githubusercontent.com/54288924/227574596-5f3c7ce1-b7c7-40da-ae6f-b4e7f80bc85b.png)

## Returning tensor immediately after self.decoder_norm() step as the model output yields an updating loss
![image](https://user-images.githubusercontent.com/54288924/227574819-b1273817-fb1a-49e5-92ab-0d514425f863.png)

Are there any special considerations to be made when training this model outside of the training script?

**Comment by aliwagdy2580** [comment]
How did you fix the code ?

**Comment by sym330** [comment]
My loss is always 0, which means the iou is also fixed

**Comment by sym330** [comment]
I put my own data set into the ade20k folder and used "python -m segm.train --log-dir seg_tiny_mask --dataset ade20k \
  --backbone vit_tiny_patch16_384 --decoder mask_transformer" for training.  I set  data.ade20k.py as follow , to detection oil.

```
class ADE20KSegmentation(BaseMMSeg):
    def __init__(self, image_size, crop_size, split, **kwargs):
        super().__init__(
            image_size,
            crop_size,
            split,
            ADE20K_CONFIG_PATH,
            **kwargs,
        )
        self.names, self.colors = utils.dataset_cat_description(ADE20K_CATS_PATH)
        self.n_cls = 2
        self.ignore_label = 255
        self.reduce_zero_label =False
```
 How is the.yml file set up in data?My data set has only one class of objects and backgrounds.
I set it as follows:
- color:
  - 127
  - 0
  - 0
  id: 0
  isthing: 0
  name: oil
- color:
  - 0
  - 0
  - 0
  id: 1
  isthing: 0
  name: background
But my losses keep going down but my mIOU stays the same. This question has been bothering me for a long time. Looking forward to your reply.

**Comment by rstrudel** [comment]
@plehman2000 did you try with the linear head?
If you still have issues, please take a look at #61 .
tldr: mmsegmentation is actively maintained and you can reach similar performances as this repository with Segmenter.

**Comment by sym330** [comment]
I tried to use a linear head, but the metrics remained the same

**Comment by sym330** [comment]
I tried to use a linear head, but the metrics remained the same

**Comment by gg22mm** [comment]
我也遇到这样的问题： https://github.com/rstrudel/segmenter/issues/68
